Test Results Summary

maythuwin1313 opened the pull request:
Closes #31645

maythuwin1313 committed:
Add summary of test results

refs #31645

maythuwin1313 committed:
Add unit test for checking summary results

refs #31645

maythuwin1313 committed:
Add function to skip comparison if base is not found

refs #31645

maythuwin1313 committed:
Add unit test to check No base in test_resultssummary

refs #31645

maythuwin1313 committed:
System exit if there is no event results and add unit test to check it

refs #31645

maythuwin1313 committed:
Remove whitespace

refs #31645

loganharbour commented on python/TestHarness/resultssummary/summary.py:
```diff
@@ -0,0 +1,107 @@
+    @staticmethod
+    def diff_table(results: TestHarnessResults, base_names: set[TestName],
+                   head_names: set[TestName]) -> Tuple[list, list]:
+        removed_names = base_names - head_names
```
Assert at the top that the arguments are the expected types

loganharbour commented on python/TestHarness/resultssummary/summary.py:
```diff
@@ -0,0 +1,107 @@
+
+    def pr(self, **kwargs) -> str:
+        results, head_names, base_names = self.pr_test_names(**kwargs)
+        if not base_names:
+            print("Skipping Test Summary Results")
+            return
```
Do we need to print this here? Isn't there already a print in pr_test_names?

loganharbour commented on python/TestHarness/tests/test_resultssummary.py:
```diff
@@ -0,0 +1,148 @@
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @unittest.skipUnless(HAS_AUTH, "Skipping because authentication is not available")
```
Similarly, we should have a simple doc string for each test method that describes what the test is testing

maythuwin1313 committed:
Add docstring into summary to explain what each function does

refs #31645

maythuwin1313 committed:
Remove duplicated print for no base in pr function

refs #31645

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3269381/), step Docs: sync website on 80cc690 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31646/site)

This comment will be updated on new commits.

maythuwin1313 committed:
Put assert at the top of functions to make sure arguments are the expected types

refs #31645

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3269375/), step Generate coverage on 80cc690 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31646/coverage/xfem)

</details>



This comment will be updated on new commits.

maythuwin1313 committed:
Add to tests build summary when test are no changed, removed or newly added

refs #31645

maythuwin1313 committed:
Update minor changes based on review comments

- Update Optional on diff_table args
- Add moose python header
- Update to include the header markdown syntax
- Update base is NoneType for checking no base available
- Update on docstring

refs #31645

maythuwin1313 committed:
Remove duplicate header syntax

refs #31645

maythuwin1313 committed:
Add additional summary result of same testnames but > head runtime threadshold and > relative runtime

refs #31645

maythuwin1313 committed:
Update all unit test functions due to additional summary result for same testnames

refs #31645

maythuwin1313 committed:
Added path to output summary

refs #31645

maythuwin1313 committed:
Add summary_output file function to display summary results in output path

refs #31645

maythuwin1313 committed:
Update docstring for summary output file and pr

refs #31645

maythuwin1313 committed:
Display summary result using print function

refs #31645

maythuwin1313 committed:
Assign head run time threadshold and relative run time rate as constant under class level

refs #31645

maythuwin1313 committed:
Add unit test for same tests with high relative runtime rate

refs #31645

maythuwin1313 committed:
Update to absolute relative runtime calculation

refs #31645

maythuwin1313 committed:
Update parseArgs to include head runtime threadshold and relative run time rate

refs #31645

maythuwin1313 committed:
Update output file to write error message or skip test

refs #31645

maythuwin1313 committed:
Update in unit test to include head runtime threadshold and relative run time rate as args instead of constant

refs #31645

maythuwin1313 committed:
Put assert on output path and add tmp path in unit test

refs #31645

maythuwin1313 committed:
Improve docstring and code formatting for readability

refs #31645

maythuwin1313 committed:
Renames due to rebase of new resulsstore

maythuwin1313 committed:
Add function for get_commit_resutls and get_event_results in summary.py

refs #31645

maythuwin1313 committed:
Sample unit test to add mock

refs #31645

moosebuild commented:
Job [Python 3.13 (Apptainer)](https://civet.inl.gov/job/3282989/) on 950cc14 : invalidated by @loganharbour

maythuwin1313 committed:
Test diff table and pr_test_names using mock data

refs #31645

maythuwin1313 committed:
Test format_table and build summary using mock data

refs #31645

maythuwin1313 committed:
Create subfunction for formatting table and rearranging code readiability

refs #31645

maythuwin1313 committed:
Test output file using mock data

refs #31645

maythuwin1313 committed:
Test pr and main using mock data

refs #31645

maythuwin1313 committed:
Add mock_init_reader return value as None

refs #31645

maythuwin1313 committed:
Remove foo testing mock

refs #31645

maythuwin1313 committed:
Update docstring in test_restulssummary

refs #31645

maythuwin1313 committed:
Update docstring for summary.py to include optional praser and subfunctions

loganharbour reviewed (changes requested):
I haven't looked at the unit tests yet. I can once these comments are addressed

maythuwin1313 committed:
Update overall doc_string and update function name

refs #31645

maythuwin1313 committed:
Update removed_names to removed_table conversion using list comprehension

refs #31645

maythuwin1313 committed:
Add event_id and out parameters in pr() in addition to kwargs and pr_test_names() instead of kwargs

refs #31645

maythuwin1313 committed:
Update summary_output to write_output in unit test to align with main function

refs #31645

maythuwin1313 committed:
Update function and docstring to wrap instead of long line

refs #31645

maythuwin1313 committed:
Update tests to include test_resultssummary

refs #31645

maythuwin1313 committed:
Remove authentication needed for database

refs #31645

maythuwin1313 committed:
Update docstring for spacing alignment and out to out_file for consistency

refs #31645

maythuwin1313 committed:
Update pr_test_names function for spacing alignment

refs #31645

maythuwin1313 committed:
Update each testname as individual list in remove table

refs #31645

maythuwin1313 committed:
Assert check to ensure table type before formatting and Update test name as code format

loganharbour reviewed (changes requested):
- We need to add command line option to disable the run time comparison, so that we can disable it if it isn't reliable or useful
- Output the runtimes with just two values after the decimal (i.e., 1.12). We don't care about printing all the extra numbers
- The runtime change should include a + or a - for if the runtime increased or decreased in the head
- The runtime changes should be sorted from increasing to decreasing order (from tests that got longer to tests that got shorter)

maythuwin1313 committed:
Add disable option for run time comparison

refs #31645

maythuwin1313 committed:
Update general format table function instead of individual format table function

refs #31645

maythuwin1313 committed:
Refractor diff_table function in summary

refs #31645

maythuwin1313 committed:
Update to construct table based on runtime comparison option and sorted table based on run time or run time rate

refs #31645

maythuwin1313 committed:
Remove whitespace

refs #31645

maythuwin1313 committed:
Update to reuturn None if added_table is empty

refs #31645

maythuwin1313 committed:
Update unit test after refactoring diff table and adding disable run time option

refs #31645

maythuwin1313 committed:
Update run time format

refs #31645

maythuwin1313 committed:
Update docstring, add assert check after refactoring diff table and adding disable run time option

refs #31645

maythuwin1313 committed:
Remove unused args and rearrange code for readiability in test_resultssummary

refs #31645

maythuwin1313 committed:
Update run time type as float

refs #31645

maythuwin1313 committed:
Update to cover cases where run time is None for added tests or same tests

refs #31645

maythuwin1313 committed:
Update docstring to include format testname, runtime, and check no run time case

refs #31645

maythuwin1313 committed:
Update correct float format to check

refs #31645

loganharbour reviewed (changes requested):
One other thing - for all comments there should be a space after the comment.

For example:

```
#This is a comment
```

should be:

```
# This is a comment
```

maythuwin1313 committed:
Format comments, remove unrelated comments, update correct type of return in comments

refs #31645

maythuwin1313 committed:
Update to show list of new tests and their runtimes, it's not related to disable runtime comparison option

refs #31645

maythuwin1313 committed:
Remove to print message in console since it's written in ouput file

refs #31645

maythuwin1313 committed:
Update Header Names in summary table for simplicity

refs #31645

loganharbour reviewed (changes requested):
Tests look great! This is just about done.

Tomorrow - we should have a single test that tries to read a PR (from the database, if available) and otherwise this is good.

maythuwin1313 committed:
Update table header description

refs #31645

maythuwin1313 committed:
Update requirement description in tests for resultssummary

refs #31645

maythuwin1313 committed:
Add single test that tries to read a PR (from the database, if available)

refs #31645

maythuwin1313 committed:
Update testPRReadDatabase test to assert the case if database not found

refs #31645

maythuwin1313 committed:
Update testPRLive to read pr from database

refs #31645

maythuwin1313 committed:
Add run time column for removed tests to display in summary table

refs #31645

maythuwin1313 committed:
Update unit test to add run time column for removed tests

refs #31645

moosebuild commented:
Job [WIP Results summary](https://civet.inl.gov/job/3299198/), step Results summary on 2fa4760 wanted to post the following:

## Framework test summary

### Removed tests
None
### Added tests
None
### Run time changes
| Test                                                                                    |   Base (s) |   Head (s) | +/-     |
|-----------------------------------------------------------------------------------------|------------|------------|---------|
| `postprocessors/side_extreme_value.g/aux_elemental`                                     |       0.78 |       1.39 | +77.34% |
| `misc/respect_petsc_options.take_vec_mat_types`                                         |       0.85 |       1.29 | +52.34% |
| `misc/intermittent_failure.intermittent_failure`                                        |       0.92 |       1.4  | +51.76% |
| `kernels/kernel_precompute.adtest`                                                      |       0.8  |       1.22 | +51.57% |
| `postprocessors/interface_value.interface_average_variable_value_postprocessor_fv_test` |       0.85 |       1.28 | +50.99% |
| `kernels/scalar_kernel_constraint.bipass`                                               |       0.82 |       1.23 | +50.50% |
## Modules test summary

### Removed tests
None
### Added tests
None
### Run time changes
None
